### PR TITLE
Moved action events into the toolbar event services and hooked them up

### PR DESCRIPTION
**Pull request by linusnorton**

### CODE REVIEW

File: projects/media-viewer/src/lib/toolbar/main-toolbar/left-pane/left-pane.component.ts [hunk 1]
```diff
@@ -9,26 +7,28 @@ import { ToolbarButtonVisibilityService } from '../../toolbar-button-visibility.
...
+  public pageNumber = 1;
 
   constructor(
-    public readonly toolbarEventsService: ToolbarEventService,
+    public readonly toolbarEvents: ToolbarEventService,
     public readonly toolbarButtons: ToolbarButtonVisibilityService
   ) {}
 
+  public ngOnInit(): void {
+    this.toolbarEvents.setCurrentPage.subscribe(pageNumber => this.setCurrentPage(pageNumber));
```
Comment (russellwardink) [inline projects/media-viewer/src/lib/toolbar/main-toolbar/left-pane/left-pane.component.ts]: cool, but don’t forget to unsubscribe in ngOnDestroy

suggestion:
    ngOnDestroy(): void {
    // Clean up any subscriptions that we may have
    for (const subscription of this.subscriptions) {
      subscription.unsubscribe();
    }
  }

File: projects/media-viewer/src/lib/toolbar/main-toolbar/middle-pane/middle-pane.component.ts [hunk 2]
```diff
@@ -10,27 +9,23 @@ import { ToolbarButtonVisibilityService } from '../../toolbar-button-visibility.
...
   constructor(
-    public readonly toolbarButtons: ToolbarButtonVisibilityService
+    public readonly toolbarButtons: ToolbarButtonVisibilityService,
+    public readonly toolbarEvents: ToolbarEventService
   ) {}
 
-  zoom(zoomFactor: number) {
-    this.zoomEvent.next(new ZoomOperation(zoomFactor));
+  zoom(zoomFactor: string) {
+    this.toolbarEvents.zoom.next(+zoomFactor);
```
Comment (russellwardink) [inline projects/media-viewer/src/lib/toolbar/main-toolbar/middle-pane/middle-pane.component.ts]: just a query… why the +

File: projects/media-viewer/src/lib/toolbar/main-toolbar/search-bar/search-bar.component.ts [hunk 3]
```diff
@@ -21,11 +19,16 @@ export class SearchBarComponent {
   haveResults = false;
 
   constructor(
-    public readonly toolbarButtons: ToolbarButtonVisibilityService
+    public readonly toolbarButtons: ToolbarButtonVisibilityService,
+    public readonly toolbarEvents: ToolbarEventService
   ) {}
 
+  public ngOnInit(): void {
+    this.toolbarEvents.searchResultsCount.subscribe(results => this.setSearchResultsCount(results));
```
Comment (russellwardink) [inline projects/media-viewer/src/lib/toolbar/main-toolbar/search-bar/search-bar.component.ts]: as before, don’t forget in these instances to unsubscribe in the ngOnDestroy

File: projects/media-viewer/src/lib/viewers/unsupported-viewer/unsupported-viewer.component.ts [hunk 4]
```diff
@@ -1,23 +1,24 @@
...
-  set downloadOperation(operation: DownloadOperation) {
-    if (operation) {
-      this.downloadLink.nativeElement.click();
-    }
+  constructor(
+    public readonly toolbarEvents: ToolbarEventService
+  ) {}
+
+  public ngOnInit(): void {
+    this.toolbarEvents.download.subscribe(() => this.downloadLink.nativeElement.click());
```
Comment (russellwardink) [inline projects/media-viewer/src/lib/viewers/unsupported-viewer/unsupported-viewer.component.ts]: don’t forget your ngOnDestroy to unsubscribe.

File: projects/media-viewer/src/lib/media-viewer.component.html [hunk 5]
```diff
@@ -2,49 +2,20 @@
...
-                   [printOperation]="actionEvents?.print | async"
-                   [rotateOperation]="actionEvents?.rotate | async"
-                   [setCurrentPage]="actionEvents?.setCurrentPage | async"
-                   [changePageByDelta]="actionEvents?.changePageByDelta | async"
-                   [currentPageChanged]="currentPageChanged">
+                   [downloadFileName]="downloadFileName">
     </mv-pdf-viewer>
     <mv-image-viewer *ngIf="contentType === 'image'"
                      [url]="url"
                      [annotationSet]="showAnnotations ? annotationSet : null"
```
[the remark this answers is not in the document]
Comment (linusnorton) [reply projects/media-viewer/src/lib/media-viewer.component.html]: It seems like a pretty legitimate input. Taleb and I did discuss creating a AnnotationSetRepository (Service) but all the changes I was making relate to the toolbar actions so I didn't do it yet

File: projects/media-viewer/src/lib/toolbar/main-toolbar/middle-pane/middle-pane.component.ts (hunk 2 above)
Comment (linusnorton) [reply projects/media-viewer/src/lib/toolbar/main-toolbar/middle-pane/middle-pane.component.ts]: casts it to a number

File: projects/media-viewer/src/lib/toolbar/main-toolbar/search-bar/search-bar.component.ts (hunk 3 above)
Comment (linusnorton) [reply projects/media-viewer/src/lib/toolbar/main-toolbar/search-bar/search-bar.component.ts]: added

File: projects/media-viewer/src/lib/toolbar/main-toolbar/left-pane/left-pane.component.ts [hunk 6]
```diff
@@ -20,6 +19,10 @@ export class ToolbarLeftPaneComponent implements OnInit {
     this.toolbarEvents.setCurrentPage.subscribe(pageNumber => this.setCurrentPage(pageNumber));
   }
 
+  public ngOnDestroy(): void {
+    this.toolbarEvents.setCurrentPage.unsubscribe();
```
Comment (russellwardink) [inline projects/media-viewer/src/lib/toolbar/main-toolbar/left-pane/left-pane.component.ts]: not sure this is going to work…. you need to call unsubscribe on the Subscription object the subscribe returned

File: projects/media-viewer/src/lib/viewers/unsupported-viewer/unsupported-viewer.component.ts (hunk 4 above)
Comment (linusnorton) [reply projects/media-viewer/src/lib/viewers/unsupported-viewer/unsupported-viewer.component.ts]: done

File: projects/media-viewer/src/lib/toolbar/main-toolbar/left-pane/left-pane.component.ts (hunk 1 above)
Comment (linusnorton) [reply projects/media-viewer/src/lib/toolbar/main-toolbar/left-pane/left-pane.component.ts]: done

File: projects/media-viewer/src/lib/toolbar/main-toolbar/left-pane/left-pane.component.ts (hunk 6 above)
Comment (linusnorton) [reply projects/media-viewer/src/lib/toolbar/main-toolbar/left-pane/left-pane.component.ts]: oh....

**Pull request merged**